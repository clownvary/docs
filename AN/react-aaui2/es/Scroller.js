import _classCallCheck from 'babel-runtime/helpers/classCallCheck';
import _possibleConstructorReturn from 'babel-runtime/helpers/possibleConstructorReturn';
import _inherits from 'babel-runtime/helpers/inherits';
import { PureComponent } from 'react';
import { func, number } from 'prop-types';

var Scroller = function (_PureComponent) {
  _inherits(Scroller, _PureComponent);

  function Scroller() {
    _classCallCheck(this, Scroller);

    var _this = _possibleConstructorReturn(this, _PureComponent.call(this));

    _this.state = {
      scrolled: false
    };

    _this.onScroll = function () {
      var threshold = _this.props.threshold;

      var scrollHeight = Math.max(document.documentElement.scrollTop, window.pageYOffset, document.body.scrollTop);

      _this.setState(function () {
        return {
          scrolled: scrollHeight > threshold
        };
      });
    };

    _this.boundOnScroll = _this.onScroll.bind(_this);
    return _this;
  }

  Scroller.prototype.componentDidMount = function componentDidMount() {
    window.addEventListener('scroll', this.boundOnScroll);
  };

  Scroller.prototype.componentWillUnmount = function componentWillUnmount() {
    window.removeEventListener('scroll', this.boundOnScroll);
  };

  Scroller.prototype.render = function render() {
    var scrolled = this.state.scrolled;
    var children = this.props.children;


    return children(scrolled);
  };

  return Scroller;
}(PureComponent);

Scroller.propTypes = {
  children: func,
  threshold: number
};
Scroller.defaultProps = {
  threshold: 50
};
export default Scroller;