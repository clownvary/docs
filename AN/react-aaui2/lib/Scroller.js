'use strict';

exports.__esModule = true;

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');

var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);

var _inherits2 = require('babel-runtime/helpers/inherits');

var _inherits3 = _interopRequireDefault(_inherits2);

var _react = require('react');

var _propTypes = require('prop-types');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Scroller = function (_PureComponent) {
  (0, _inherits3.default)(Scroller, _PureComponent);

  function Scroller() {
    (0, _classCallCheck3.default)(this, Scroller);

    var _this = (0, _possibleConstructorReturn3.default)(this, _PureComponent.call(this));

    _this.state = {
      scrolled: false
    };

    _this.onScroll = function () {
      var threshold = _this.props.threshold;

      var scrollHeight = Math.max(document.documentElement.scrollTop, window.pageYOffset, document.body.scrollTop);

      _this.setState(function () {
        return {
          scrolled: scrollHeight > threshold
        };
      });
    };

    _this.boundOnScroll = _this.onScroll.bind(_this);
    return _this;
  }

  Scroller.prototype.componentDidMount = function componentDidMount() {
    window.addEventListener('scroll', this.boundOnScroll);
  };

  Scroller.prototype.componentWillUnmount = function componentWillUnmount() {
    window.removeEventListener('scroll', this.boundOnScroll);
  };

  Scroller.prototype.render = function render() {
    var scrolled = this.state.scrolled;
    var children = this.props.children;


    return children(scrolled);
  };

  return Scroller;
}(_react.PureComponent);

Scroller.propTypes = {
  children: _propTypes.func,
  threshold: _propTypes.number
};
Scroller.defaultProps = {
  threshold: 50
};
exports.default = Scroller;
module.exports = exports['default'];