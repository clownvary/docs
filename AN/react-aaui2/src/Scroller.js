import { PureComponent } from 'react'
import { func, number } from 'prop-types'

export default class Scroller extends PureComponent {
  static propTypes = {
    children: func,
    threshold: number,
  }

  static defaultProps = {
    threshold: 50,
  }

  constructor() {
    super()

    this.boundOnScroll = this.onScroll.bind(this)
  }

  state = {
    scrolled: false,
  }

  componentDidMount() {
    window.addEventListener('scroll', this.boundOnScroll)
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.boundOnScroll)
  }

  onScroll = () => {
    const { threshold } = this.props
    const scrollHeight = Math.max(
      document.documentElement.scrollTop,
      window.pageYOffset,
      document.body.scrollTop,
    )

    this.setState(() => ({
      scrolled: scrollHeight > threshold,
    }))
  }

  render() {
    const { scrolled } = this.state
    const { children } = this.props

    return children(scrolled)
  }
}
