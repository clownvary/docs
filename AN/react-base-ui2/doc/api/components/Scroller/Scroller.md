## Scroller 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
horizontal|bool|true|false|&ensp;
vertical|bool|true|false|&ensp;