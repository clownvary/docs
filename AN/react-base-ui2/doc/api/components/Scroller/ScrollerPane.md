## ScrollerPane 



### Prop Types
Property | Type | Default Value | Is Required | Description
:--- | :--- | :--- | :--- | :---
horizontal|bool|false|false|&ensp;
vertical|bool|false|false|&ensp;
group|string|''|false|&ensp;