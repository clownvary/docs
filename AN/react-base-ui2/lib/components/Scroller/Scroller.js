'use strict';Object.defineProperty(exports, "__esModule", { value: true });var _getPrototypeOf = require('babel-runtime/core-js/object/get-prototype-of');var _getPrototypeOf2 = _interopRequireDefault(_getPrototypeOf);var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);var _createClass2 = require('babel-runtime/helpers/createClass');var _createClass3 = _interopRequireDefault(_createClass2);var _possibleConstructorReturn2 = require('babel-runtime/helpers/possibleConstructorReturn');var _possibleConstructorReturn3 = _interopRequireDefault(_possibleConstructorReturn2);var _inherits2 = require('babel-runtime/helpers/inherits');var _inherits3 = _interopRequireDefault(_inherits2);var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _propTypes = require('prop-types');
var _uniqueId = require('lodash/uniqueId');var _uniqueId2 = _interopRequireDefault(_uniqueId);
var _isFunction = require('lodash/isFunction');var _isFunction2 = _interopRequireDefault(_isFunction);
var _classnames = require('classnames');var _classnames2 = _interopRequireDefault(_classnames);
var _ScrollerPane = require('./ScrollerPane');var _ScrollerPane2 = _interopRequireDefault(_ScrollerPane);
var _dom = require('../../utils/dom');function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

/**
                                                                                                                                     * Default PropTypes of Scroller.
                                                                                                                                     */
var ScrollerPropTypes = {
  horizontal: _propTypes.bool,
  vertical: _propTypes.bool };


/** Default Props for Scroller */
var ScrollerProps = {
  horizontal: true,
  vertical: true };var


Scroller = function (_React$PureComponent) {(0, _inherits3.default)(Scroller, _React$PureComponent);




  function Scroller() {(0, _classCallCheck3.default)(this, Scroller);var _this = (0, _possibleConstructorReturn3.default)(this, (Scroller.__proto__ || (0, _getPrototypeOf2.default)(Scroller)).call(this));

    _this.name = (0, _uniqueId2.default)('an-scroller-');
    _this.onContentResize = _this.onContentResize.bind(_this);return _this;
  }(0, _createClass3.default)(Scroller, [{ key: 'getContentSize', value: function getContentSize()

    {
      if (this.contentPane) {
        return this.contentPane.getSize();
      }

      return null;
    } }, { key: 'onContentResize', value: function onContentResize(

    e) {
      if (this.header) {
        var w = e.scrollbarSize.width;
        if (w > 0) {
          this.header.style.marginRight = w - 1 + 'px';
          (0, _dom.addClass)(this.header, 'rightscroll');
        } else {
          this.header.style.marginRight = '';
          (0, _dom.removeClass)(this.header, 'rightscroll');
        }
      }

      if (this.band) {
        this.band.style.marginBottom = '';

        var h = e.scrollbarSize.height;
        if (h > 0) {
          this.band.style.marginBottom = h - 1 + 'px';
          (0, _dom.addClass)(this.band, 'bottomscroll');
        } else {
          (0, _dom.removeClass)(this.band, 'bottomscroll');
        }
      }var

      onContentResize = this.props.onContentResize;
      if ((0, _isFunction2.default)(onContentResize)) {
        onContentResize(e);
      }
    } }, { key: 'render', value: function render()

    {var _this2 = this;var _props =










      this.props,horizontal = _props.horizontal,vertical = _props.vertical,style = _props.style,className = _props.className,corner = _props.corner,header = _props.header,band = _props.band,content = _props.content,onScroll = _props.onScroll;

      var classes = (0, _classnames2.default)(
      'an-scroller an-border-single',
      className);


      return (
        _react2.default.createElement('div', { className: classes, style: style },

          header && _react2.default.createElement('div', null,
            _react2.default.createElement('div', { className: 'an-scroller__header-wrapper' },
              band && _react2.default.createElement('div', { className: 'an-scroller__corner', ref: function ref(c) {_this2.corner = c;} }, corner),
              _react2.default.createElement('div', { className: 'an-scroller__header', ref: function ref(c) {_this2.header = c;} },
                _react2.default.createElement(_ScrollerPane2.default, {
                    horizontal: false,
                    vertical: false,
                    group: this.name },

                  header)))),






          _react2.default.createElement('div', { className: 'an-scroller__body-wrapper' },

            band && _react2.default.createElement('div', { className: 'an-scroller__band', ref: function ref(c) {_this2.band = c;} },
              _react2.default.createElement(_ScrollerPane2.default, {
                  horizontal: false,
                  vertical: false,
                  group: this.name },

                band)),



            _react2.default.createElement('div', { className: 'an-scroller__content', ref: function ref(c) {_this2.content = c;} },
              _react2.default.createElement(_ScrollerPane2.default, {
                  ref: function ref(c) {_this2.contentPane = c;},
                  horizontal: horizontal,
                  vertical: vertical,
                  group: this.name,
                  onResize: this.onContentResize,
                  onScroll: onScroll,
                  master: true },

                content)))));





    } }]);return Scroller;}(_react2.default.PureComponent);Scroller.displayName = 'Scroller';Scroller.defaultProps = ScrollerProps;Scroller.propTypes = ScrollerPropTypes;exports.default =



Scroller;module.exports = exports['default'];