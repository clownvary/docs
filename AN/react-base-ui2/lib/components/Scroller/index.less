.an-scroller {
  display: flex;
  flex-direction: column;
  overflow: hidden;

  &__header-wrapper {
    display: flex;
    flex-direction: row;
    height: auto;
  }

  &__corner {
    flex: none;
  }

  &__header {
    flex: 1;
    width: 0;
  }

  &__body-wrapper {
    display: flex;
    flex: 1;
    flex-direction: row;
    height: 0;
  }

  &__band {
    width: auto;
  }

  &__content {
    flex: 1;
    width: 0;
    height: 100%;
  }
}

.an-scroller-pane {
  width: 100%;
  height: 100%;
  overflow: hidden;
  outline: none;
}

.bottomscroll {
  .an-scroller-pane {
    border-bottom: solid 1px #ddd;
  }
}

.rightscroll {
  .an-scroller-pane {
    border-right: solid 1px #ddd;
  }
}

.an-border-single {
  border: solid 1px lightgray;
}
