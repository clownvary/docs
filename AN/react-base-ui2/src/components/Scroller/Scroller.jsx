import React from 'react';
import { bool } from 'prop-types';
import uniqueId from 'lodash/uniqueId';
import isFunction from 'lodash/isFunction';
import classNames from 'classnames';
import ScrollerPane from './ScrollerPane';
import { addClass, removeClass } from '../../utils/dom';

/**
 * Default PropTypes of Scroller.
 */
const ScrollerPropTypes = {
  horizontal: bool,
  vertical: bool
};

/** Default Props for Scroller */
const ScrollerProps = {
  horizontal: true,
  vertical: true
};

class Scroller extends React.PureComponent {
  static displayName = 'Scroller';
  static defaultProps = ScrollerProps;
  static propTypes = ScrollerPropTypes;

  constructor() {
    super();
    this.name = uniqueId('an-scroller-');
    this.onContentResize = this.onContentResize.bind(this);
  }

  getContentSize() {
    if (this.contentPane) {
      return this.contentPane.getSize();
    }

    return null;
  }

  onContentResize(e) {
    if (this.header) {
      const w = e.scrollbarSize.width;
      if (w > 0) {
        this.header.style.marginRight = `${w - 1}px`;
        addClass(this.header, 'rightscroll');
      } else {
        this.header.style.marginRight = '';
        removeClass(this.header, 'rightscroll');
      }
    }

    if (this.band) {
      this.band.style.marginBottom = '';

      const h = e.scrollbarSize.height;
      if (h > 0) {
        this.band.style.marginBottom = `${h - 1}px`;
        addClass(this.band, 'bottomscroll');
      } else {
        removeClass(this.band, 'bottomscroll');
      }
    }

    const { onContentResize } = this.props;
    if (isFunction(onContentResize)) {
      onContentResize(e);
    }
  }

  render() {
    const {
      horizontal,
      vertical,
      style,
      className,
      corner,
      header,
      band,
      content,
      onScroll
    } = this.props;

    const classes = classNames(
      'an-scroller an-border-single',
      className
    );

    return (
      <div className={classes} style={style}>
        {
          header && <div>
            <div className="an-scroller__header-wrapper">
              {band && <div className="an-scroller__corner" ref={(c) => { this.corner = c; }}>{corner}</div>}
              <div className="an-scroller__header" ref={(c) => { this.header = c; }}>
                <ScrollerPane
                  horizontal={false}
                  vertical={false}
                  group={this.name}
                >
                  {header}
                </ScrollerPane>
              </div>
            </div>
          </div>
        }

        <div className="an-scroller__body-wrapper">
          {
          band && <div className="an-scroller__band" ref={(c) => { this.band = c; }}>
            <ScrollerPane
              horizontal={false}
              vertical={false}
              group={this.name}
            >
              {band}
            </ScrollerPane>
          </div>
        }
          <div className="an-scroller__content" ref={(c) => { this.content = c; }}>
            <ScrollerPane
              ref={(c) => { this.contentPane = c; }}
              horizontal={horizontal}
              vertical={vertical}
              group={this.name}
              onResize={this.onContentResize}
              onScroll={onScroll}
              master
            >
              {content}
            </ScrollerPane>
          </div>
        </div>
      </div>
    );
  }

}

export default Scroller;
