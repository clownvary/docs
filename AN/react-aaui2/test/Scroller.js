import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Scroller from 'Scroller'

const setup = () => {
  const Hello = () => <h1>123</h1>

  const Component = <Scroller>{() => <Hello />}</Scroller>
  const wrapper = mount(Component)

  return {
    wrapper,
    Component,
  }
}

it('should render without errors', () => {
  const { wrapper } = setup()

  expect(wrapper).toBeTruthy()
})

it("should render right thing and don't change unexpected", () => {
  const { Component } = setup()

  const tree = renderer.create(Component).toJSON()

  expect(tree).toMatchSnapshot()
})

it('should fix menu when scroll down over thresholdHeight', () => {
  const { wrapper } = setup()

  document.body.scrollTop = 500

  wrapper.instance().onScroll()

  expect(wrapper.state('scrolled')).toBe(true)
  wrapper.unmount()
})
