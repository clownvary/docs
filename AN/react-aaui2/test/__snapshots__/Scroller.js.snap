// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render right thing and don't change unexpected 1`] = `
<h1>
  123
</h1>
`;
